File: app/models/note.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Text, func
from sqlalchemy.orm import relationship
from app.db.database import Base

class Note(Base):
    __tablename__ = "notes"
    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    category_id = Column(Integer, ForeignKey("categories.id", ondelete="SET NULL"), nullable=True)
    event_id = Column(Integer, ForeignKey("events.id", ondelete="SET NULL"), nullable=True)
    title = Column(String(255), nullable=False)
    content = Column(Text, nullable=False)
    color_code = Column(String(7))
    created_at = Column(DateTime, server_default=func.now())

    category = relationship("Category", back_populates="notes")
    event = relationship("Event", back_populates="notes")
    owner = relationship("User", back_populates="notes")

File: Mobile/TODO_App/app/src/main/java/com/ysr_klc/todo_app/model/Note.kt
package com.ysr_klc.todo_app.model

import com.google.gson.annotations.SerializedName

data class Note(
    @SerializedName("id") val id: Int,
    @SerializedName("user_id") val userId: Int,
    @SerializedName("category_id") val categoryId: Int?,
    @SerializedName("event_id") val eventId: Int?,

    @SerializedName("title") val title: String,
    @SerializedName("content") val content: String,

    @SerializedName("color_code") val colorCode: String?,
    @SerializedName("created_at") val createdAt: String?
)